OutOfSchool/OutOfSchool.BusinessLogic/Services/Database/WorkshopService.cs

```diff
@@ -631,6 +648,121 @@ public async Task<SearchResult<WorkshopCard>> GetNearestByFilter(WorkshopFilter
         return result;
     }
 
+    /// <inheritdoc/>
+    /// <exception cref="InvalidOperationException">If the region admin is not found in the database.</exception>
+    public async Task<SearchResult<WorkshopDto>> FetchByFilterForAdmins(WorkshopFilterAdministration filter = null)
+    {
+        logger.LogInformation("Started retrieving Workshops by filter for admins.");
+        filter = filter ?? new WorkshopFilterAdministration();
+
+        var predicate = await PredicateBuildForAdminds(filter);
+
+        var workshops = await workshopRepository.Get(
+                skip: filter.From,
+                take: filter.Size,
+                includeProperties: includingPropertiesForMappingDtoModel,
+                whereExpression: predicate,
+                asNoTracking: true)
+            .ToListAsync()
+            .ConfigureAwait(false);
+
+        var workshopsDTO = mapper.Map<List<WorkshopDto>>(workshops);
+        var workshopsCount = workshops.Count;
+
+        logger.LogInformation("Retrieved {WorkshopsCount} matching recods by filter for admins.", workshopsCount);
+
+        var result = new SearchResult<WorkshopDto>()
+        {
+            TotalAmount = workshopsCount,
+            Entities = workshopsDTO,
+        };
+
+        return result;
+    }
+
+    private async Task<Expression<Func<Workshop, bool>>> PredicateBuildForAdminds(WorkshopFilterAdministration filter)
+    {
+        var predicate = PredicateBuilder.True<Workshop>();
+
+        // TODO: add condition for united territorial community admin
+        if (currentUserService.IsMinistryAdmin())
+        {
+           await ApplyFilterForMinistryAdmin(filter);
+        }
+        else if (currentUserService.IsRegionAdmin())
+        {
+            var tempPredicate = await ApplyFilterForRegionAdmin(filter);
+            predicate = predicate.And(tempPredicate);
+        }
+
+        if (!string.IsNullOrEmpty(filter.SearchString))
+        {
+            var tempPredicate = PredicateBuilder.False<Workshop>();
+
+            foreach (var word in filter.SearchString.Split(' ', ',', StringSplitOptions.RemoveEmptyEntries))
```

Comment:
I suppose these splitters should not be here.
It looks like a part of a 'Search' business logic.
Why dot is not here ? Are there other word separators like `-` or `;` ?

Reply:
I built the filtering based on the analogy with the filtering logic of providers , where a search string is also used. Thank you, I indeed didn't consider other separators, I will fix it.

Reply:
I will create a  service for handling the search string in a separate pr.